src/basic/stat-util.c

```diff
@@ -451,11 +451,19 @@ int statx_fallback(int dfd, const char *path, int flags, unsigned mask, struct s
 }
 
 int xstatfsat(int dir_fd, const char *path, struct statfs *ret) {
+        _cleanup_free_ char *nested = NULL;
         _cleanup_close_ int fd = -EBADF;
 
         assert(dir_fd >= 0 || dir_fd == AT_FDCWD);
         assert(ret);
 
+        nested = path_join(path, "a");
+        if (!nested)
+                return -ENOMEM;
+
+        /* Make sure we trigger any automounts by trying to access a file inside the given path. */
+        (void) faccessat(dir_fd, nested, F_OK, 0);
```

Comment:
We can make it configurable if needed but it seems good to do this by default. It's very hard to consider when writing code that a directory might be automounted and I think that usually we want to check something of the automounted directory instead of the mountpoint.

A good example is that we broke `bootctl status` and `systemctl kexec` by not doing this.